python/desc/imsim/bleed_trails.py

```diff
@@ -0,0 +1,175 @@
+"""
+Simple bleed trail implementation to apply to "eimages" just prior to
+the camera readout code.
+"""
+import lsst.afw.image as afw_image
+import lsst.afw.detection as afw_detect
+
+__all__ = ['apply_channel_bleeding', 'bleed_eimage',
+           'find_channels_with_saturation', 'bleed_channel']
+
+
+def apply_channel_bleeding(gs_interpreter, full_well):
+    """
+    Apply charge bleeding to eimages with channels that have pixels
+    above full well.
+
+    Parameters
+    ----------
+    gs_interpreter: lsst.sims.GalSimInterface.GalSimInterpreter
+        The object that is actually drawing the images
+    full_well: int
+        The pixel full well/saturation value in electrons.
+    """
+    for gs_image in gs_interpreter.detectorImages.values():
+        bleed_eimage(afw_image.ImageF(gs_image.array), full_well)
+
+
+def bleed_eimage(eimage, full_well, midline_stop=True):
+    """
+    Apply the bleed charge algorithm to an eimage.
+
+    Parameters
+    ----------
+    eimage: lsst.afw.image.ImageF
+        The "eimage" containing the image data in units of electrons per
+        pixel.  This is the image prior to electronic readout (i.e.,
+        conversion to ADU, addition of bias, dark current, crosstalk, etc.).
+        For LSST CCDs, the eimages have parallel transfer directions along
+        the x-axis, hence channels correspond to rows in eimages.
+    full_well: int
+        The pixel full well/saturation value in electrons.
+    midline_stop: bool [True]
+        Flag to treat the midline of the sensor as a bleed stop.
+
+    Returns
+    -------
+    afw.image.ImageF: This is the input eimage object with the charge
+        bleeding applied.
+    """
+    channels = find_channels_with_saturation(eimage, full_well)
+    imarr = eimage.getArray()
+    for ypix in channels:
+        if midline_stop:
+            xmid = imarr.shape[1]//2
+            imarr[ypix, :xmid] = bleed_channel(imarr[ypix, :xmid], full_well)
+            imarr[ypix, xmid:] = bleed_channel(imarr[ypix, xmid:], full_well)
+        else:
+            imarr[ypix, :] = bleed_channel(imarr[ypix, :], full_well)
+    return eimage
+
+
+def find_channels_with_saturation(eimage, full_well):
```

Comment:
I find "channel" a bit ambiguous.  What does it mean, amp channel, pixel, etc.  Would 'column' be better?so: find_columns_with_saturation etc.

Reply:
I wanted to avoid `column` because it implies being parallel to the y-axis, whereas for LSST eimages, the channels are along the x-axis in the Camera Coordinate System.  Also, the `lsst.afw.detection` code has the relevant dimensions of its `span`s indexed in x.  Finally, "channel" seems to be the correct technical term for this feature of CCDs, e.g., see the Janesick book.